File: samples/perl/05wc/wc.pl
use warnings;
use strict;

my $len = scalar(@ARGV);
if ($len > 1) {
    print STDERR "Error: Expected to receive $0 [<file>]\n";
    exit(1);
}

my $lines = 0;
while (<>) {
    $lines++;
}

print "$lines\n";

File: samples/cs/05wc/wc.cs
using System;
using System.IO;

namespace LineCount {
    class Exec {
        static void Main(string[] args) {
            if (args.Length > 1) {
                Console.Error.WriteLine("Error: Expected to receive at most 1 argument ./wc [<file>]");
                System.Environment.Exit(1);
            }

            int lines = 0;
            if (args.Length == 1) {
                Console.SetIn(new StreamReader(args[0]));
            }

            while (Console.ReadLine() != null) {
                lines++;
            }

            Console.WriteLine(lines);
        }
    }
}
